File: cpp/2235_sum/sum.cpp
/*
 * @Date: 2023-08-19
 * @LastEditors: 854284842@qq.com
 * @LastEditTime: 2023-08-19
 * @FilePath: /algorithm/cpp/2235_sum/sum.cpp
 */

#include <cassert>
#include <tuple>
#include <vector>

class Solution {
 public:
  int sum(int num1, int num2) { return num1 + num2; }
};

int main() {
  std::vector<std::tuple<int, int, int>> tests{
      {12, 5, 17},
      {-10, 4, -6},
  };

  for (auto &[num1, num2, ans] : tests) {
    assert(Solution().sum(num1, num2) == ans);
  }
}

File: golang/2235_sum/sum.go
/*
 * @Date: 2023-08-19
 * @LastEditors: 854284842@qq.com
 * @LastEditTime: 2023-08-19
 * @FilePath: /algorithm/golang/2235_sum/sum.go
 */

// Package main ...
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func sum(num1 int, num2 int) int {
	return num1 + num2
}

func main() {
	tests := []struct {
		num1 int
		num2 int
		ans  int
	}{
		{12, 5, 17},
		{-10, 4, -6},
	}

	for _, item := range tests {
		assert.Equal(&testing.T{}, item.ans, sum(item.num1, item.num2))
	}
}
